src/d3a/models/area/event_dispatcher.py

```diff
@@ -215,8 +215,13 @@ def _delete_past_agents(self, area_agent_member):
                     if hasattr(agent, "engines"):
                         for engine in agent.engines:
                             del engine.forwarded_offers
+                            del engine.offer_age
+                            del engine.trade_residual
+                            del engine.ignored_offers
```

Comment:
Out of interest: How much memory did this change save?

Reply:
Cannot say exactly, because this was mostly an extra security layer. The real change that actually forced the IAAs to be deleted was this: https://github.com/gridsingularity/d3a/pull/739/files#diff-d67f7f839ba3365d818a4d178e4b487fL79 . I can run a (not so quick) test though if you are interested, ping me if you would like to. Thanks!